pkgs/development/python-modules/tensorrt/default.nix

```diff
@@ -0,0 +1,58 @@
+{ lib
+, python
+, buildPythonPackage
+, autoPatchelfHook
+, unzip
+, cudaPackages
+, cudatoolkit
+}:
+
+let
+  pyVersion = "${lib.versions.major python.version}${lib.versions.minor python.version}";
+in
+buildPythonPackage rec {
+  pname = "tensorrt";
+  version = cudaPackages.tensorrt.version;
+
+  src = cudaPackages.tensorrt.src;
+
+  format = "wheel";
+  # We unpack the wheel ourselves because of the odd packaging.
+  dontUseWheelUnpack = true;
+
+  nativeBuildInputs = [
+    unzip
+    autoPatchelfHook
+  ];
+
+  preUnpack = ''
+    mkdir -p dist
+    tar --strip-components=2 -xf "$src" --directory=dist \
+      "TensorRT-${version}/python/tensorrt-${version}-cp${pyVersion}-none-linux_x86_64.whl"
+  '';
+
+  sourceRoot = ".";
+
+  runtimeDependencies = [
+    cudaPackages.tensorrt
+  ];
+
+  propagatedBuildInputs = [
+    cudatoolkit
```

Comment:
I assume these are for autopatchelf. Move to `buildInputs`.

`propagatedBuildInputs` are really only used for python packages and largely because they don't have headers where we could write the rpaths; we avoid `propagatedBuildInputs`, because one can soon  shoot one's feet with multiple instances of the same package, especially when they're as obscure and heavy as cuda/cudnn

Reply:
Moved to `buildInputs`.